packages/shopping/src/services/user-management.service.ts

```diff
@@ -9,17 +9,22 @@ import {repository} from '@loopback/repository';
 import {HttpErrors} from '@loopback/rest';
 import {securityId, UserProfile} from '@loopback/security';
 import {PasswordHasherBindings} from '../keys';
-import {UserWithPassword, User} from '../models';
+import {NodeMailer, User, UserWithPassword} from '../models';
 import {Credentials, UserRepository} from '../repositories';
 import {PasswordHasher} from './hash.password.bcryptjs';
 import _ from 'lodash';
+import {EmailService} from './email.service';
+import {v4 as uuidv4} from 'uuid';
+import {subtractDates} from '../utils';
 
 export class UserManagementService implements UserService<User, Credentials> {
   constructor(
     @repository(UserRepository)
     public userRepository: UserRepository,
     @inject(PasswordHasherBindings.PASSWORD_HASHER)
     public passwordHasher: PasswordHasher,
+    @inject('services.EmailService')
```

Comment:
Preferably, we shouldn't need to know the exact binding key when doing dependency injection (separation of responsibilities). So either we derive the binding key from the class _or_ create a binding key in `keys.ts` like the password hasher above this.
```suggestion
    @service(EmailService)
```

Reply:
I think this needs another PR to fix since the pattern already exists. See `product.controller.ts` and `user-management.controller.ts`. The case for password hasher is a unique one since this is a custom service that doesn't use default binding attributes of the framework